src/analyses/variable-sensitivity/abstract_environment.cpp

```diff
@@ -468,3 +499,211 @@ std::vector<abstract_object_pointert> eval_operands(
+    if(result.is_nil() || result.is_false())
+      return result;
+  }
+  return true_exprt();
+}
+
+exprt assume_or(
+  abstract_environmentt &env,
+  const exprt &expr,
+  const namespacet &ns)
+{
+  exprt or_result = false_exprt();
+  auto or_expr = to_or_expr(expr);
+  for(auto const &operand : or_expr.operands())
+  {
+    auto result = env.do_assume(operand, ns);
+    if(result.is_nil())
+      return result;
+    if(result.is_true())
+      or_result = true_exprt();
+  }
+  return or_result;
+}
+
+exprt assume_eq(
+  abstract_environmentt &env,
+  const exprt &expr,
+  const namespacet &ns)
+{
+  auto const &equal_expr = to_binary_expr(expr);
+
+  auto left = env.eval(equal_expr.lhs(), ns);
+  auto right = env.eval(equal_expr.rhs(), ns);
+
+  if(left->is_top() || right->is_top())
+    return nil_exprt();
+  if(!is_value(left) || !is_value(right))
+    return nil_exprt();
+
+  auto meet = left->meet(right);
+
+  if(meet->is_bottom())
+    return false_exprt();
+
+  if(is_lvalue(equal_expr.lhs()))
+    env.assign(equal_expr.lhs(), meet, ns);
+  if(is_lvalue(equal_expr.rhs()))
+    env.assign(equal_expr.rhs(), meet, ns);
+  return true_exprt();
+}
+
+exprt assume_noteq(
+  abstract_environmentt &env,
+  const exprt &expr,
+  const namespacet &ns)
+{
+  auto const &notequal_expr = to_binary_expr(expr);
+
+  auto left = env.eval(notequal_expr.lhs(), ns);
+  auto right = env.eval(notequal_expr.rhs(), ns);
+
+  if(left->is_top() || right->is_top())
+    return nil_exprt();
+  if(!is_value(left) || !is_value(right))
+    return nil_exprt();
+
+  auto meet = left->meet(right);
+
+  if(meet->is_bottom())
+    return true_exprt();
+
+  return false_exprt();
+}
+
+struct left_and_right_valuest
+{
+  abstract_value_pointert left;
+  abstract_value_pointert right;
+
+  constant_interval_exprt left_interval() const
+  {
+    return left->to_interval();
+  }
+  constant_interval_exprt right_interval() const
+  {
+    return right->to_interval();
+  }
+
+  bool are_good() const
+  {
+    return left != nullptr && right != nullptr;
+  }
+};
+
+left_and_right_valuest eval_operands_as_values(
+  abstract_environmentt &env,
+  const exprt &expr,
+  const namespacet &ns)
+{
+  auto const &relationship_expr = to_binary_expr(expr);
+
+  auto left = env.eval(relationship_expr.lhs(), ns);
+  auto right = env.eval(relationship_expr.rhs(), ns);
+
+  if(left->is_top() || right->is_top())
+    return {};
+
+  return {as_value(left), as_value(right)};
+}
+
+exprt assume_less_than(
+  abstract_environmentt &env,
+  const exprt &expr,
+  const namespacet &ns)
+{
+  auto operands = eval_operands_as_values(env, expr, ns);
+  if(!operands.are_good())
+    return nil_exprt();
+
+  auto left_lower = operands.left_interval().get_lower();
+  auto right_upper = operands.right_interval().get_upper();
+
+  auto reduced_le_expr =
+    binary_relation_exprt(left_lower, expr.id(), right_upper);
+  return env.eval(reduced_le_expr, ns)->to_constant();
+}
+
+exprt assume_greater_than(
```

Comment:
One of the symmetries that does work is `(a > b) == (b < a)` so if you want you can implement one of greater and less and simply recurse, flipping the arguments, to the other.